### PR TITLE
Chore: split models.py into separate files for ease of maintenance

### PR DESCRIPTION
**Pull request by danted22**
**Description**

Split models.py into separate files for ease of maintenance.

### CODE REVIEW

File: API_DOCS.md [hunk 1]
```diff
@@ -164,8 +164,8 @@ For full instructions check the [`backend README`](./backend/README.md)
 
 **Returns**: An object containing:
   - Success (Boolean) - a success value.
-  - users (List) - List of users matching the query.
```
Comment (shirblc) [inline API_DOCS.md]: Can you revert these? Everything should be capitalised in the types in the docs; the few that aren't should be fixed

File: models/base_models.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+from sqlalchemy.orm import (
+    DeclarativeBase,
+)
+
+
+class BaseModel(AsyncAttrs, DeclarativeBase):
+    pass
+
+
+HugModelType = TypeVar("HugModelType", bound=BaseModel, covariant=True)
```
Comment (shirblc) [inline models/base_models.py]: Two thoughts:
1. `base_models` feels like an inaccurate name considering there's only one model in it. Maybe base/common/something along those lines would be a better fit, what do you think?
2. if we're grouping everything together, which is a good idea, we can probably put the models in db.py here too (unless that's too much of a problem with circular imports)

File: models/schemas/permissions.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# SOFTWARE.from models.models import BaseModel, DumpedModel
+
+
+from sqlalchemy import Integer, String
+from sqlalchemy.orm import Mapped, mapped_column
+
+from models.base_models import BaseModel, DumpedModel
+
+
+class Permission(BaseModel):
```
Comment (shirblc) [inline models/schemas/permissions.py]: I think we can just group it into roles, no? It's part of the same thing

File: models/schemas/threads.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+from sqlalchemy import ForeignKey, Integer, and_, case, false, func, or_, select, true
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import Mapped, column_property, mapped_column, relationship
+
+
+import json
+
+
+class Thread(BaseModel):
```
Comment (shirblc) [inline models/schemas/threads.py]: Same here, shouldn't this be in messaging? It's basically just a mapping table for that

**Review by shirblc — CHANGES_REQUESTED** [review]
Looks good overall, definitely a lot easier to work with! Just a few small comments.

Only other thing that bugged me was the order of imports is backwards in most of the files, but since you already said you're going to add `isort`, it doesn't really warrant a comment 😄

File: models/base_models.py (hunk 2 above)
Comment (danted22) [reply models/base_models.py]: Do you mean CoreSAHModel? That's the only one that looks like a model to me but I might be mistaken

File: models/schemas/threads.py (hunk 4 above)
Comment (danted22) [reply models/schemas/threads.py]: makes sense.

File: models/schemas/permissions.py (hunk 3 above)
Comment (danted22) [reply models/schemas/permissions.py]: Yep, done.

File: API_DOCS.md (hunk 1 above)
Comment (danted22) [reply API_DOCS.md]: Apologies. Overzealous find replace 😉

File: models/base_models.py (hunk 2 above)
Comment (danted22) [reply models/base_models.py]: have renamed it common - happy to move CoreSAHModel in there

**Review by danted22 — COMMENTED** [review]
ok, covered everything apart from CoreSAHModel - once you confirm that's the only one I can move it and we should be done with this PR assuming tests are ok

File: models/base_models.py (hunk 2 above)
Comment (shirblc) [reply models/base_models.py]: Exactly, meant `CoreSAHModel` and the dataclass. That way db.py keeps just the database class, which is what it should've been in the first place

Comment (danted22) [reply models/base_models.py]: cool, moved!

File: tests/data_models.py [hunk 5]
```diff
@@ -1,23 +1,22 @@
...
-    Role,
     NotificationSub,
+    Post,
+    Report,
+    Role,
+    User,
 )
 from models.db import SendADatabase
-
+from models.schemas.messages import Thread
```
Comment (shirblc) [inline tests/data_models.py]: Is there a reason these two aren't imported from `models`?

**Review by shirblc — DISMISSED** [review]

File: tests/data_models.py (hunk 5 above)
Comment (danted22) [reply tests/data_models.py]: Cos vscode is crap at auto refactoring? 🤣

I'll fix

**Review by shirblc — APPROVED** [review]

**Pull request merged**